pkg/compactor/compactor_test.go

```diff
@@ -0,0 +1,249 @@
+				restoreDir, err = ioutil.TempDir("/tmp", "restore-")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				restoreOpts.Config.RestoreDataDir = restoreDir
+
+				restoreOpts.BaseSnapshot = compactedSnapshot
+				restoreOpts.DeltaSnapList = deltaSnapList
+
+				rstr := restorer.NewRestorer(store, logger)
+
+				err = rstr.RestoreAndStopEtcd(*restoreOpts)
+
+				Expect(err).ShouldNot(HaveOccurred())
+				err = utils.CheckDataConsistency(testCtx, restoreOpts.Config.RestoreDataDir, keyTo, logger)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+		})
+		Context("with defragmention not allowed", func() {
+			AfterEach(func() {
+				_, err := os.Stat(restoreDir)
+				if err != nil {
+					os.RemoveAll(restoreDir)
+				}
+				store.Delete(*compactedSnapshot)
+			})
+			It("should restore from compacted snapshot", func() {
+				restoreOpts.Config.MaxFetchers = 4
+
+				// Fetch latest snapshots
+				baseSnapshot, deltaSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				restoreOpts.BaseSnapshot = baseSnapshot
+				restoreOpts.DeltaSnapList = deltaSnapList
+
+				// Take the compacted full snapshot with defragmnetation not allowed
+				_, err = cptr.Compact(restoreOpts, false)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Check if the compacted full snapshot is really present
+				snapList, err := store.List()
+				Expect(err).ShouldNot(HaveOccurred())
+
+				compactedSnapshot = snapList[len(snapList)-1]
+				fi, err := os.Stat(path.Join(compactedSnapshot.Prefix, compactedSnapshot.SnapDir, compactedSnapshot.SnapName))
+				Expect(err).ShouldNot(HaveOccurred())
+
+				size := fi.Size()
+				Expect(size).ShouldNot(BeZero())
+			})
+			It("should create a snapshot", func() {
+				restoreOpts.Config.MaxFetchers = 4
+
+				// Fetch latest snapshots
+				baseSnapshot, deltaSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				restoreOpts.BaseSnapshot = baseSnapshot
+				restoreOpts.DeltaSnapList = deltaSnapList
+
+				// Take the compacted full snapshot with defragmnetation not allowed
+				_, err = cptr.Compact(restoreOpts, false)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				// Check if the compacted full snapshot is really present
+				compactedSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+				Expect(err).ShouldNot(HaveOccurred())
+				fi, err := os.Stat(path.Join(compactedSnapshot.Prefix, compactedSnapshot.SnapDir, compactedSnapshot.SnapName))
+				Expect(err).ShouldNot(HaveOccurred())
+
+				size := fi.Size()
+				Expect(size).ShouldNot(BeZero())
+
+				// Restore from the compacted snapshot
+				restoreDir, err = ioutil.TempDir("/tmp", "restore-")
+				Expect(err).ShouldNot(HaveOccurred())
+
+				restoreOpts.Config.RestoreDataDir = restoreDir
+
+				restoreOpts.BaseSnapshot = compactedSnapshot
+				restoreOpts.DeltaSnapList = deltaSnapList
```

Comment:
Thanks for adding the restoration check!

However, this doesn't check if `miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)` is returning the compacted snapshot. Could you please make that call and add `Expect`s to be absolutely sure that the restoration will happen from the compacted snapshot.